tests/contrib/io/matplotlib/test_matplotlib_writer.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2018-2019 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import pytest
+
+from kedro.contrib.io.matplotlib import MatplotlibWriter
+from kedro.io import DataSetError
+
+matplotlib.use("Agg")  # Disable interactive mode
+
+
+class TestMatplotlibWriter:
+    def test_simgle_image(self, tmp_path):
+        # generate a plot
+        # pylint: disable=unsubscriptable-object,useless-suppression
+        plt.plot(np.random.rand(1, 5)[0], np.random.rand(1, 5)[0])
+
+        # write and compare
+        trusted_filepath = tmp_path / "image_we_expect.png"
+        plt.savefig(str(trusted_filepath))
+
+        experimental_filepath = tmp_path / "image_we_write.png"
+        plot_writer = MatplotlibWriter(filepath=str(experimental_filepath))
+        plot_writer.save(plt)
+        plt.close()
+
+        assert trusted_filepath.read_bytes() == experimental_filepath.read_bytes()
```

Comment:
I would avoid using terms like `trusted` and `experimental`, as they only make sense to people with the business knowledge behind them. It probably makes a lot of sense to a DE, but not immediately obvious to other software engineers. Standard terminology like "actual" and "expected" is imo much clearer at conveying the message in tests and making them more readable (I don't have to think to translate).

Reply:
Agree, I reused the original variables, but will update them :)